src/test/java/org/testng/internal/issue2426/SuiteTest.java

```diff
@@ -0,0 +1,25 @@
+package org.testng.internal.issue2426;
+
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.xml.SuiteXmlParser;
+import org.testng.xml.XmlSuite;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import test.SimpleBaseTest;
+
+public class SuiteTest extends SimpleBaseTest {
+    @Test (description = "ISSUE-2426")
+    public void testGetFactoryMethodParamsInfo() throws IOException {
+        final SuiteXmlParser parser = new SuiteXmlParser();
+        final String testSuite = "src/test/resources/xml/github2426/test-suite.xml";
```

Comment:
Can we consider using the `XmlSuite` and `XmlTest` object so that the test is by itself fully self contained instead of using the `xml` files ?